org.eclipse.xtext/src/org/eclipse/xtext/workspace/ISourceFolder.java

```diff
@@ -20,4 +32,28 @@
 	 * separator. It will never be null.
 	 */
 	URI getPath();
+
+	/**
+	 * @param uri
+	 *            to check
+	 * @return true iff the given {@link URI} is a sub element of the {@link URI} of this {@link ISourceFolder}.
+	 */
+	default boolean contains(URI uri) {
```

Comment:
They are unused. Since it is super easy to mimic this functionality with extension methods; I am not sure we need this as an API.

Reply:
> They

The method `contains` is used twice: `FileProjectConfig.xtend` and `EclipseProjectConfig.xtend`.

> Since it is super easy to mimic this functionality with extension methods

When possible, we use Java instead of Xtend.